theano/sandbox/cuda/basic_ops.py

```diff
@@ -2614,6 +2618,75 @@ def perform(self, node, inp, out_):
                 o[j] = x[i]
             out[0] = o
 
+    def c_code(self, node, name, inputs, outputs, sub):
+        x, idx = inputs
+        out, = outputs
+        fail = sub['fail']
+        if node.inputs[0].ndim not in [2, 3]:
+            raise NotImplementedError("This case does not have C code yet.")
+        if node.inputs[1].dtype != 'int64':
+            raise Exception("Index should have dtype int64. Check this node make_node().")
+        return """
+        int max_threads=512;
+        //take(idx, 0, out, "raise", max_threads);
+        PyObject * out_ = NULL;
+        PyObject * ret = NULL;
+        PyObject * args = PyTuple_New(5);
+        PyObject * zero = PyInt_FromLong(0);
+        PyObject * max = PyInt_FromLong(max_threads);
+        PyObject * raise = PyString_FromString("raise");
+        if(args == NULL || zero == NULL || max == NULL || raise == NULL){
+            %(fail)s;
+        }
+
+        out_ = (PyObject *) %(out)s;
+        if (out_ == NULL)
+            out_ = Py_None;
+        Py_INCREF(out_);
+        Py_INCREF(%(idx)s);
+
+        // In the error handling, I don't decref args as I don't know
+        // how it decref not assigned item.
+
+        if(PyTuple_SetItem(args, 0, (PyObject *) %(idx)s)){
+            Py_DECREF(%(idx)s); Py_DECREF(zero); Py_DECREF(out_);
+            Py_DECREF(raise); Py_DECREF(max); Py_DECREF(out_);
+            %(fail)s;
+        }
+        if(PyTuple_SetItem(args, 1, zero)){
+            Py_DECREF(%(idx)s); Py_DECREF(zero); Py_DECREF(out_);
+            Py_DECREF(raise); Py_DECREF(max); Py_DECREF(out_);
+            %(fail)s;
+        }
+        if(PyTuple_SetItem(args, 2, out_)){
+            Py_DECREF(%(idx)s); Py_DECREF(zero); Py_DECREF(out_);
+            Py_DECREF(raise); Py_DECREF(max); Py_DECREF(out_);
+            %(fail)s;
+        }
+        if(PyTuple_SetItem(args, 3, raise)){
+            Py_DECREF(%(idx)s); Py_DECREF(zero); Py_DECREF(out_);
+            Py_DECREF(raise); Py_DECREF(max); Py_DECREF(out_);
+            %(fail)s;
+        }
+        if(PyTuple_SetItem(args, 4, max)){
+            Py_DECREF(%(idx)s); Py_DECREF(zero); Py_DECREF(out_);
+            Py_DECREF(raise); Py_DECREF(max); Py_DECREF(out_);
+            %(fail)s;
+        }
```

Comment:
I think this code would be clearer (and shorter) if you used [Py_BuildValue()](https://docs.python.org/2/c-api/arg.html#c.Py_BuildValue) like:
```
Py_BuildValue("OiOsi", %(idx)s, 0, %(out)s == NULL ? Py_None, %(out)s, "raise", max_threads);
```